frontend/src/js/components/Plotter.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import Plot from './Plot.jsx'
+
+var Plotter = React.createClass({
+    initDrop() {
+        $('.cell-stack-bottom').droppable({
+            over: (event, ui) => {
+                $(event.target).addClass('plotter-to-top');
+            },
+            out: (event, ui) => {
+                $(event.target).removeClass('plotter-to-top');
+            }
+        })
+    },
+    addPlot(event, ui) {
+        let variable = null;
+        let graphics_method_parent = null;
+        let graphics_method = null;
+        let template = null;
+        switch (ui.draggable.attr('data-type')) {
+            case 'variable':
+                this.props.addPlot(ui.draggable.attr('data-name'), graphics_method_parent, graphics_method, template, this.props.row, this.props.col);
+                break
+            case 'graphics_method':
+                this.props.addPlot(variable, ui.draggable.attr('data-parent'), ui.draggable.attr('data-name'), template, this.props.row, this.props.col);
+                break
+            case 'template':
+                this.props.addPlot(variable, graphics_method_parent, graphics_method, ui.draggable.attr('data-name'), this.props.row, this.props.col);
+                break
+            default:
+                break
+        }
+        $('.cell-stack-bottom').removeClass('plotter-to-top');
+    },
+    componentDidMount(){
+        this.initDrop();
+        $('#add-plot-' + this.props.row + this.props.col).droppable({
+            tolerance: 'pointer',
+            hoverClass: 'plot-hover',
+            drop: this.addPlot
+        })
+    },
+    componentDidUpdate(){
+        this.initDrop();
+    },
+    render() {
+        return (
+            <div className='cell-stack-bottom'>
+                <div className='plotter-plots'>
+                    {(() => {
+                        let plotters = [];
+                        for (var i = 0; i < this.props.cell.plots.length; i++) {
+                            let plot = this.props.cell.plots[i];
+                            let plot_name = 'plot' + this.props.row + this.props.col + i;
```

Comment:
Changing the sheet changes the plots that are populated throughout the gui because everything is populated using currently selected sheet in the redux object. So when sheet changes plots gets changed as well because plots is part of each sheet object.